packages/react-client/src/ReactFlightClient.js

```diff
@@ -960,6 +980,22 @@ function initializeModelChunk<T>(chunk: ResolvedModelChunk<T>): void {
         return;
       }
     }
+
+    if (__DEV__ && isReactElementOrArrayLike(value)) {
+      const debugInfo = chunk._debugInfo.splice(0);
+      if (value._debugInfo) {
+        // $FlowFixMe[method-unbinding]
+        value._debugInfo.push.apply(value._debugInfo, debugInfo);
```

Comment:
Maybe this should be unshifted to the beginning because if this just is like a Server Component that renders something that itself was reused. Then the outer one is this Server Component and the thing it renders is the inner one.

So I think pushing leads to wrong order.